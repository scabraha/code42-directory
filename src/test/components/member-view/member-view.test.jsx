import React from 'react';
import renderer from 'react-test-renderer';
import MemberView from '../../../components/member-view/member-view';

jest.mock('../../../components/member-list', () => 'MemberList');

describe('Member View', () => {
  it('Renders the view with an organization and loading status and ensures it renders correctly', () => {
    const tree = renderer.create(
      <MemberView organization="testOrgnization" teamMemberRequestStatus="LOADING" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders the view with an organization and success status and ensures it renders correctly', () => {
    const tree = renderer.create(
      <MemberView organization="testOrgnization" teamMemberRequestStatus="SUCCESS" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders the view with an organization and failure status and ensures it renders correctly', () => {
    const tree = renderer.create(
      <MemberView organization="testOrgnization" teamMemberRequestStatus="FAILURE" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders the view with an organization and not started status and ensures it renders correctly', () => {
    const tree = renderer.create(
      <MemberView organization="testOrgnization" teamMemberRequestStatus="NOT_STARTED" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

