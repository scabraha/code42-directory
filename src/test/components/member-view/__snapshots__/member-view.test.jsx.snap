// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member View Renders the view with an organization and failure status and ensures it renders correctly 1`] = `
<div
  className="member-view"
>
  <div
    className="member-view-text md-subheading-2 md-text-center"
  >
    <div>
      <i
        className="material-icons"
      >
        
      </i>
    </div>
    Member retrieval failure
  </div>
</div>
`;

exports[`Member View Renders the view with an organization and loading status and ensures it renders correctly 1`] = `
<div
  className="member-view"
>
  <div
    className="member-view-text md-subheading-2 md-text-center"
  >
    <div>
      <i
        className="loading-spinner material-icons"
      >
        
      </i>
    </div>
    Retrieving members for 
    testOrgnization
  </div>
</div>
`;

exports[`Member View Renders the view with an organization and not started status and ensures it renders correctly 1`] = `
<div
  className="member-view"
/>
`;

exports[`Member View Renders the view with an organization and success status and ensures it renders correctly 1`] = `
<div
  className="member-view"
>
  <MemberList />
</div>
`;
